backend/app/testbot/data_loader.py

```
@@ -1,13 +1,162 @@
+    if "discountedPrice" in clean_product:
+        try:
+            clean_product["discountedPrice"] = float(clean_product["discountedPrice"])
+        except (ValueError, TypeError):
+            pass
+    
+    # Clean product name
+    if "name" in clean_product:
+        clean_product["name"] = html.unescape(clean_product["name"]).strip()
+    
+    # Ensure URL is valid
+    if "url" in clean_product and clean_product["url"]:
```

Ruff (0.8.2)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)